uaclient/cli.py

```diff
@@ -1372,22 +1377,23 @@ def action_api(args, *, cfg):
 
 
 @assert_root
-@assert_lock_file("pro auto-attach")
-def action_auto_attach(args, *, cfg):
-    if cfg.is_attached:
-        raise exceptions.AlreadyAttachedOnPROError()
-
-    skip_auto_attach = actions.should_disable_auto_attach(cfg)
-    if skip_auto_attach:
-        return 0
-
-    instance = actions.get_cloud_instance(cfg)
-
+def action_auto_attach(args, *, cfg: config.UAConfig) -> int:
     try:
-        actions.auto_attach(cfg, instance)
+        _full_auto_attach(
+            FullAutoAttachOptions(),
+            cfg=cfg,
+            mode=event_logger.EventLoggerMode.CLI,
+        )
     except exceptions.UrlError:
         event.info(messages.ATTACH_FAILURE.msg)
         return 1
+    except api_exceptions.AutoAttachDisabledError:
+        return 0
+    except exceptions.AlreadyAttachedError as e:
+        raise exceptions.AlreadyAttachedOnPROError() from e
```

Comment:
`AlreadyAttachedOnPRO` only existed to change the return code to 0. And we onnly changed the return code to 0 in that case because we used to call this cli command in the on-boot ua-auto-attach.service.

Now that this is really only a CLI function to be called by a human, I think we should return 1 for both "already attached" and for "auto attach disabled". What do you think @lucasmoura @renanrodrigo @CalvoM 

If we follow this proposal then we don't need any special handling for these exceptions, and we can delete `AlreadyAttachedOnPRO`

Reply:
Thanks for the context here. My main concern was actually the message change, as the exceptions contain different error messages [1] that are later shown to the user. If that change is acceptable, we could simplify this and probably drop `AlreadyAttachedOnPRO` and its associated message, as we only use it here.

[1] https://github.com/canonical/ubuntu-advantage-client/blob/a5f3f0eb06a4ae6732d17fda4de5b11f298ed72b/uaclient/messages.py#L590-L602/blob/a5f3f0eb06a4ae6732d17fda4de5b11f298ed72b/uaclient/messages.py#L590-L602